elastic/kibana
# [Fleet] fix for apm not found in epr + prerelease flag

## Summary

Fixes https://github.com/elastic/kibana/issues/146312

Fall back to bundled packages when loading package archive.

Also fixed an issue with prerelease flag caused by https://github.com/elastic/kibana/pull/144431

Loading prerelease setting on Add/Edit integration page, so that it works well for cases where the url doesn't contain a concrete version like APM

Steps to verify:
- Start kibana locally, make sure that APM bundled package is there in `x-pack/plugins/fleet/target/bundled_packages`
- Load http://localhost:5601/julia/app/integrations/detail/apm/overview (with default "beta integrations" off flag)
- Expect that 8.6.0 version is showing up (or 8.7.0 depending on the build)
- When clicking on Add Elastic APM, expect 8.6.0 to be loaded successfully

<img width="1773" alt="image" src="https://user-images.githubusercontent.com/90178898/206482128-634ca823-bb21-46a9-a8b9-fc81d5dc126d.png">

Note: there is still a UI issue with loading assets on Overview page if the package is not in EPR.

<img width="1747" alt="image" src="https://user-images.githubusercontent.com/90178898/206478275-ee8e9494-234f-44c8-88d9-a91c791b0196.png">

- switch "beta integrations" on on Integrations page
- Open APM overview page again
- expect `8.7.0-preview` version with the selector to change to last GA

<img width="1100" alt="image" src="https://user-images.githubusercontent.com/90178898/206480986-2ed05b12-49e4-4c88-97c6-6cdf12f819da.png">


### Checklist
